src/lib/rules/axe/axe.ts

```diff
@@ -11,7 +11,7 @@ import { AxeResults, Result as AxeResult, NodeResult as AxeNodeResult } from 'ax
 
 import { readFileAsync } from '../../utils/misc';
 import { debug as d } from '../../utils/debug';
-import { IAsyncHTMLElement, ITraverseEnd, IRule, IRuleBuilder, Severity } from '../../types'; // eslint-disable-line no-unused-vars
+import { IAsyncHTMLElement, IRule, IRuleBuilder, Severity, ITraverseStart } from '../../types'; // eslint-disable-line no-unused-vars
```

Comment:
Now that I think about it, we have a problem switching to `traverse::start` with `chrome`.
We might want to create a custom event for evaluating scripts and each `connector` can fire it when it is more appropriate.
Maybe something like `canevaluate`?

@sonarwhal/contributors what do you think?

Reply:
@molant I haven't encounter any issue with `axe` using `chrome` so far. What's the issue with using `traverse::start` in chrome?
> Now that I think about it, we have a problem switching to traverse::start with chrome.

That way maybe we can have a more clear message to show when evaluate is running, like `Running scripts...`, rather than hanging at `Traversing the DOM` when it's actually evaluating.
> Maybe something like canevaluate?

Reply:
>  I haven't encounter any issue with axe using chrome so far.

The problem is that we will be telling people to `evaluate` scripts on `traverse::start`. Those scripts could modify the DOM and alter the results. That's why I'd rather do all of that at the end of the process for `chrome`. With `jsdom` we don't have that problem because it is a separate process.

Reply:
@molant I put it back as `traverse::end` for now before we decide whether or not we should have a custom event.

Reply:
For reference, the discussion about this is now in https://github.com/sonarwhal/sonar/issues/497